frontends/p4/sideEffects.cpp

```diff
@@ -279,8 +317,11 @@ class DismantleExpression : public Transform {
 
     // We don't want to compute the full read/write set here so we
     // overapproximate it as follows: all declarations that occur in
-    // an expression.  TODO: this could be made more precise, perhaps
-    // using LocationSets.
+    // an expression.
+
+    // TODO: this could be made more precise, perhaps using LocationSets.
+    // TODO: surely there are other passes that compute reads/writes?  Perhaps
```

Comment:
Unfortunately the pass that computes reads/writes depends on this one being run first. The reason is that you cannot compute reads if you don't know what side-effects occur first...